cardshifter-server/src/test/java/com/cardshifter/server/main/ServerConnectionTest.java

```diff
@@ -252,5 +256,43 @@ public void testUsersQueryNotLoggedIn() throws IOException, InterruptedException
 		// There shouldn't be a UserStatusMessage for client2
 		assertNoMessage(client1);
 	}
+
+	@Test(timeout = 5000)
+	public void testNoOfflineMessageIfNotLoggedIn() throws IOException, InterruptedException {
+		/* There's a race condition between a possible UserStatusMessage (if the test were to fail) and the
+		 * message sent by the server in response to the request from assertNoMessage. Therefore wait until
+		 * the server has processed the disconnection before continuing.
+		 *
+		 * See https://github.com/Cardshifter/Cardshifter/pull/410#discussion-diff-40526060 for discussion.
+		 */
+
+		shutdown();
```

Comment:
This test is not using the `@Before` method, it's just immediately running the `@After` method. I looked at better solutions to this and think [using inner classes](http://stackoverflow.com/a/13017356/2073469) might be the simplest approach. I don't want to put the test in a completely different class as it shares much of the code with the other tests in this suite.

There are other solutions in the question linked above, but I think they look much less clean.

Any opinions on this before I start refactoring?